### PR TITLE
feat: add XDEFI Wallet

### PR DESCRIPTION
**Pull request by loxator**
### Description
Added XDEFI Wallet as an option to show up on the modal, and upon clicking it navigate to XDEFI's website.

Not sure about the package version, I have set it with 2.0.0, please let me know if that needs to be changed.

### Checklist
- [ ] The version field in `package.json` of the package you have made changes in is incremented following [semantic versioning](https://semver.org/) and using alpha release tagging
- [ ] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works
- [x] I have run `yarn file-check`, `yarn type-check` & `yarn build` to confirm there are not any associated errors
- [ ] This PR passes the Circle CI checks

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

### CODE REVIEW

File: packages/xdefi/package.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+  "name": "@web3-onboard/xdefi",
+  "version": "2.0.0",
```
Comment (Adamj1232) [inline packages/xdefi/package.json:3]:
```suggestion
  "version": "2.0.0-alpha.1",
```

File: packages/demo/package.json [hunk 2]
```diff
@@ -47,6 +47,7 @@
     "@web3-onboard/walletconnect": "^2.2.2",
     "@web3-onboard/enkrypt": "^2.0.0",
     "@web3-onboard/mew-wallet": "^2.0.0",
+    "@web3-onboard/xdefi": "^2.0.0",
```
Comment (Adamj1232) [inline packages/demo/package.json:50]:
```suggestion
    "@web3-onboard/xdefi": "^2.0.0-alpha.1",
```

**Comment by Adamj1232** [comment]
@loxator a few comments around versioning but looks good! 
Can you also complete the PR checklist for SDK/Wallet Modules wallets using the internal demo package?

**Comment by loxator** [comment]
@Adamj1232 tried to test with injected 

- [ ] send transaction - fails with `Uncaught Error: 0xxxx-xxx-xxx-xxx-d9 is not a valid API key`, tried it with MM as well, same result
- [x] switch chains
- [x] sign message - had to change chain to polygon as `0x4` goerli isn't supported by us
- [x] sign typed message - had to change chain to polygon as `0x4` goerli isn't supported by us
- [x] disconnect


Regarding Tests with demo app (SDK), how do I go about doing this? using the `window.xfi.ethereum` object once its injected or is there some other way?

File: packages/xdefi/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  return () => {
+    return {
+      label: 'XDEFI Wallet',
+      injectedNamespace: 'xfi',
+      checkProviderIdentity: ({ provider }: { provider: any }) => {
+        !!provider && !!provider['isXDEFI']
+      },
+      getIcon: async () => (await import('./icon.js')).default,
+      getInterface: async () => {
+        const provider = window.xfi?.ethereum
```
Comment (Adamj1232) [inline packages/xdefi/src/index.ts:17]: Unfortunately for compatibility issues across bundlers and compilers we cannot use optional chaining.
```suggestion
       let provider
        if (window.xfi && window.xfi.ethereum) {
          provider = window.xfi.ethereum
        }
```

File: packages/xdefi/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        if (!provider) {
+          const newWindow = window.open(
+            'https://install.xdefi.io/?utm_source=web3Onboard&utm_medium=organic&utm_campaign=xdefi.io&utm_id=xdefi.io',
+            '_blank',
+            'noopener noreferrer'
+          )
+          if (newWindow) newWindow.opener = null
+          throw new Error('Please install XDEFI Wallet to use this wallet')
+        } else {
+          return { provider: createEIP1193Provider(window.xfi) }
```
Comment (Adamj1232) [inline packages/xdefi/src/index.ts:27]: Do we want to use `window.xfi.ethereum` here? 
```suggestion
          return { provider: createEIP1193Provider(window.xfi.ethereum) }
```

Comment (loxator) [reply packages/xdefi/src/index.ts:30]: Good catch!

**Comment by loxator** [comment]
@Adamj1232 Thanks for the review! I have committed your suggestions and tested the `Send Transaction` bit as well which worked. It should be good to merge now 👍

**Review by Adamj1232 — APPROVED** [review]
Awesome! Thank you for the PR and collaboration on it!

**Pull request merged**